src/Control/Effect/State.hs

```diff
@@ -33,57 +38,86 @@ instance Effect (State s) where
 --
 --   prop> snd (run (runState a get)) == a
 get :: (Member (State s) sig, Carrier sig m) => m s
-get = send (Get ret)
+get = send (Get pure)
 
 -- | Project a function out of the current state value.
 --
 --   prop> snd (run (runState a (gets (applyFun f)))) == applyFun f a
 gets :: (Member (State s) sig, Carrier sig m) => (s -> a) -> m a
-gets f = send (Get (ret . f))
+gets f = send (Get (pure . f))
 
 -- | Replace the state value with a new value.
 --
 --   prop> fst (run (runState a (put b))) == b
 --   prop> snd (run (runState a (get <* put b))) == a
 --   prop> snd (run (runState a (put b *> get))) == b
 put :: (Member (State s) sig, Carrier sig m) => s -> m ()
-put s = send (Put s (ret ()))
+put s = send (Put s (pure ()))
 
 -- | Replace the state value with the result of applying a function to the current state value.
 --   This is strict in the new state; if you need laziness, use @get >>= put . f@.
 --
 --   prop> fst (run (runState a (modify (+1)))) == (1 + a :: Integer)
-modify :: (Member (State s) sig, Carrier sig m, Monad m) => (s -> s) -> m ()
+modify :: (Member (State s) sig, Carrier sig m) => (s -> s) -> m ()
 modify f = do
-   a <- get
-   put $! f a
+  a <- get
+  put $! f a
 
 -- | Run a 'State' effect starting from the passed value.
 --
 --   prop> run (runState a (pure b)) == (a, b)
-runState :: forall s sig m a . (Carrier sig m, Effect sig) => s -> Eff (StateC s m) a -> m (s, a)
-runState s m = runStateC (interpret m) s
+runState :: s -> StateC s m a -> m (s, a)
+runState = flip runStateC
 
 -- | Run a 'State' effect, yielding the result value and discarding the final state.
 --
 --   prop> run (evalState a (pure b)) == b
-evalState :: forall s sig m a . (Carrier sig m, Effect sig, Functor m) => s -> Eff (StateC s m) a -> m a
-evalState s m = fmap snd (runStateC (interpret m) s)
+evalState :: forall s m a . Functor m => s -> StateC s m a -> m a
+evalState s = fmap snd . runState s
 
 -- | Run a 'State' effect, yielding the final state and discarding the return value.
 --
 --   prop> run (execState a (pure b)) == a
-execState :: forall s sig m a . (Carrier sig m, Effect sig, Functor m) => s -> Eff (StateC s m) a -> m s
-execState s m = fmap fst (runStateC (interpret m) s)
+execState :: forall s m a . Functor m => s -> StateC s m a -> m s
+execState s = fmap fst . runState s
 
 
 newtype StateC s m a = StateC { runStateC :: s -> m (s, a) }
+  deriving (Functor)
```

Comment:
Here, like with `ErrorC`, I’ve opted not to use `transformers`’ definitions since they require a bunch of extra fiddling—in this case, both extra constraints, and `fmap`s/`swap`s.